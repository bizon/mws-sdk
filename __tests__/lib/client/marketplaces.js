const {
  getMarketplaces,
  getMarketplacesFromRegion
} = require('../../../lib/client/marketplaces')

describe('lib.client.marketplaces', () => {
  describe('getMarketplacesFromRegion', () => {
    it('should throw if the provided region is unknown', () => {
      expect(
        () => getMarketplacesFromRegion('what')
      ).toThrow('what is not a valid MWS region')
    })

    it('should return a list of marketplaces for MWS regions', () => {
      const tests = ['eu', 'na', 'fe', 'ca', 'in', 'jp']

      for (const mwsRegion of tests) {
        const marketplaces = getMarketplacesFromRegion(mwsRegion)

        expect(marketplaces).toMatchSnapshot()
      }
    })
  })

  describe('getMarketplaces', () => {
    it('should throw if one of the requested marketplaces does not exist', () => {
      expect(
        () => getMarketplaces([
          'fr',
          'what',
          'A1PA6795UKMFR9'
        ])
      ).toThrow('what is not a valid marketplace code, ID or domain')
    })

    it('should deduplicate marketplaces', () => {
      expect(getMarketplaces([
        'fr',
        'amazon.fr',
        'A13V1IB3VIYZZH' // France
      ])).toHaveLength(1)
    })

    it('should ignore marketplaces without an API endpoint', () => {
      expect(getMarketplaces([
        'A13V1IB3VIYZZH',
        'A1805IZSGTT6HS',
        'A1F83G8C2ARO7P',
        'A1PA6795UKMFR9',
        'A1RKKUPIHCS9HS',
        'A1ZFFQZ3HTUKT9',
        'A38D8NSA03LJTC',
        'A62U237T8HV6N',
        'AFQLKURYRPEL8',
        'APJ6JRA9NG5V4',
        'AZMDEXL2RVFNN'
      ])).toMatchSnapshot()
    })

    it('should throw if the specified marketplaces are on multiple mws domains', () => {
      expect(
        () => getMarketplaces([
          'fr',
          'ae'
        ])
      ).toThrow('The specified marketplaces should all be on the same MWS domain, found mws-eu.amazonservices.com,mws.amazonservices.ae')
    })
  })
})
