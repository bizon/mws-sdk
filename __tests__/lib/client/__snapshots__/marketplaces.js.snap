// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib.client.marketplaces getMarketplaces should ignore marketplaces without an API endpoint 1`] = `
Array [
  Object {
    "code": "fr",
    "domain": "amazon.fr",
    "id": "A13V1IB3VIYZZH",
    "imagesDomain": "images-eu.ssl-images-amazon.com",
    "mwsDomain": "mws-eu.amazonservices.com",
    "name": "France",
    "region": "eu",
    "vendorId": "A1X6FK5RDHNB96",
  },
  Object {
    "code": "uk",
    "domain": "amazon.co.uk",
    "id": "A1F83G8C2ARO7P",
    "imagesDomain": "images-eu.ssl-images-amazon.com",
    "mwsDomain": "mws-eu.amazonservices.com",
    "name": "United Kingdom",
    "region": "eu",
    "vendorId": "A3P5ROKL5A1OLE",
  },
  Object {
    "code": "de",
    "domain": "amazon.de",
    "id": "A1PA6795UKMFR9",
    "imagesDomain": "images-eu.ssl-images-amazon.com",
    "mwsDomain": "mws-eu.amazonservices.com",
    "name": "Germany",
    "region": "eu",
    "vendorId": "A3JWKAKR8XB7XF",
  },
  Object {
    "code": "es",
    "domain": "amazon.es",
    "id": "A1RKKUPIHCS9HS",
    "imagesDomain": "images-eu.ssl-images-amazon.com",
    "mwsDomain": "mws-eu.amazonservices.com",
    "name": "Spain",
    "region": "eu",
    "vendorId": "A1AT7YVPFBWXBL",
  },
  Object {
    "code": "it",
    "domain": "amazon.it",
    "id": "APJ6JRA9NG5V4",
    "imagesDomain": "images-eu.ssl-images-amazon.com",
    "mwsDomain": "mws-eu.amazonservices.com",
    "name": "Italy",
    "region": "eu",
    "vendorId": "A11IL2PNWYJU7H",
  },
]
`;

exports[`lib.client.marketplaces getMarketplacesFromRegion should return a list of eu marketplaces 1`] = `
Array [
  Object {
    "code": "de",
    "domain": "amazon.de",
    "id": "A1PA6795UKMFR9",
    "imagesDomain": "images-eu.ssl-images-amazon.com",
    "mwsDomain": "mws-eu.amazonservices.com",
    "name": "Germany",
    "region": "eu",
    "vendorId": "A3JWKAKR8XB7XF",
  },
  Object {
    "code": "es",
    "domain": "amazon.es",
    "id": "A1RKKUPIHCS9HS",
    "imagesDomain": "images-eu.ssl-images-amazon.com",
    "mwsDomain": "mws-eu.amazonservices.com",
    "name": "Spain",
    "region": "eu",
    "vendorId": "A1AT7YVPFBWXBL",
  },
  Object {
    "code": "fr",
    "domain": "amazon.fr",
    "id": "A13V1IB3VIYZZH",
    "imagesDomain": "images-eu.ssl-images-amazon.com",
    "mwsDomain": "mws-eu.amazonservices.com",
    "name": "France",
    "region": "eu",
    "vendorId": "A1X6FK5RDHNB96",
  },
  Object {
    "code": "it",
    "domain": "amazon.it",
    "id": "APJ6JRA9NG5V4",
    "imagesDomain": "images-eu.ssl-images-amazon.com",
    "mwsDomain": "mws-eu.amazonservices.com",
    "name": "Italy",
    "region": "eu",
    "vendorId": "A11IL2PNWYJU7H",
  },
  Object {
    "code": "uk",
    "domain": "amazon.co.uk",
    "id": "A1F83G8C2ARO7P",
    "imagesDomain": "images-eu.ssl-images-amazon.com",
    "mwsDomain": "mws-eu.amazonservices.com",
    "name": "United Kingdom",
    "region": "eu",
    "vendorId": "A3P5ROKL5A1OLE",
  },
  Object {
    "code": "tr",
    "domain": "amazon.com.tr",
    "id": "A33AVAJ2PDY3EV",
    "imagesDomain": "images-eu.ssl-images-amazon.com",
    "mwsDomain": "mws-eu.amazonservices.com",
    "name": "Turkey",
    "region": "eu",
    "vendorId": "A1UNQM1SR2CHM",
  },
  Object {
    "code": "eg",
    "id": "ARBP9OOSHTCHU",
    "imagesDomain": "images-eu.ssl-images-amazon.com",
    "mwsDomain": "mws-eu.amazonservices.com",
    "name": "Egypt",
    "region": "eu",
  },
  Object {
    "code": "sa",
    "domain": "amazon.com.sa",
    "id": "A17E79C6D8DWNP",
    "imagesDomain": "images-eu.ssl-images-amazon.com",
    "mwsDomain": "mws-eu.amazonservices.com",
    "name": "Saudi Arabia",
    "region": "eu",
  },
]
`;
